pkg/controller/node/node_controller.go

```diff
@@ -101,50 +102,23 @@ func (r *ReconcileNode) Reconcile(request reconcile.Request) (reconcile.Result,
 	}
 
 	//TODO: Manage deletes
-	nodeNetworkStateKey := types.NamespacedName{
-		Namespace: "default",
-		Name:      request.Name,
-	}
-	// Create NodeNetworkState for this node
 	nodeNetworkState := &nmstatev1.NodeNetworkState{}
-	err = r.client.Get(context.TODO(), nodeNetworkStateKey, nodeNetworkState)
+	err = nmstate.GetNodeNetworkState(r.client, request.Name, nodeNetworkState)
 	if err != nil {
 		if !errors.IsNotFound(err) {
-			return reconcile.Result{}, fmt.Errorf("error accessing NodeNetworkState: %v", err)
-		}
-		nodeNetworkState.ObjectMeta = metav1.ObjectMeta{
-			Name:      request.Name,
-			Namespace: "default",
+			return reconcile.Result{}, fmt.Errorf("error at node reconcile accessing NodeNetworkState: %v", err)
 		}
-		nodeNetworkState.Spec = nmstatev1.NodeNetworkStateSpec{
-			NodeName: request.Name,
-		}
-		// There is no NodeNetworkState for this node let's create it
-		err = r.client.Create(context.TODO(), nodeNetworkState)
+	} else {
+		// NodeNetworkState is no suppose to be there let's delete it
+		err = nmstate.DeleteNodeNetworkState(r.client, nodeNetworkState)
```

Comment:
Maybe someone wants to define NodeNetworkState configuration and only then register a Node? I think we should just ignore this case until we hit an issue with it.

Reply:
Have removed in the refresh PR but let's do it here too.